gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/DatastoreImpl.java

```diff
@@ -180,23 +187,26 @@ public Entity add(FullEntity<?> entity) {
           throw DatastoreException.throwInvalidRequest(
               "Duplicate entity with the key %s", entity.key());
         }
-        mutationPb.addInsert(completeEntity.toPb());
+        mutationsPb.add(com.google.datastore.v1beta3.Mutation.newBuilder()
+            .setInsert(completeEntity.toPb()).build());
       } else {
         Preconditions.checkArgument(entity.hasKey(), "entity %s is missing a key", entity);
-        mutationPb.addInsertAutoId(entity.toPb());
+        mutationsPb.add(com.google.datastore.v1beta3.Mutation.newBuilder()
+            .setInsert(entity.toPb()).build());
       }
     }
-    DatastoreV1.CommitResponse commitResponse = commitMutation(mutationPb);
-    Iterator<DatastoreV1.Key> allocatedKeys =
-        commitResponse.getMutationResult().getInsertAutoIdKeyList().iterator();
+    com.google.datastore.v1beta3.CommitResponse commitResponse = commitMutation(mutationsPb);
```

Comment:
It turns out that mutation results only return a key when the key is incomplete (and null otherwise), so it looks like we'll need to keep the if/else statement.